src/components/schedule-visualizer/index.tsx

```diff
@@ -214,9 +216,35 @@ export default function ScheduleVisualizer(props: ScheduleVisualizerProps): Reac
     return () => clearInterval(interval);
   }, [props.trajManager, curMapFloorLayer, trajAnimDuration]);
 
+  function determineMinZoom(width: number, height: number): number {
+    if (Math.abs(width) >= Math.abs(height)) {
+      // Right now baseWidth and baseZoom are set to 50 and 4
+      // We might need to change this in future if a range of
+      // values is given
+      const baseWidth = 50;
+      const baseZoom = 4;
```

Comment:
Based on my research, zoom level 0 in leaflet means 1:1 scale of the image, i.e. a image of 1024x1024 would be shown as 1024x1024. Each zoom level means a x2 focus, so at zoom level 1 the image would be 2048x2048, and at -1 it would be 512x512 (but there appears to be some bugs in negative zooms and its not officially supported by leaflet).

Take the office demo for example, its width is roughly 25.8. On my desktop, the zoom leaflet calculates that will fit the map onto the viewport is 6 (which is 64x zoom resulting in ~1650 pixels). If I reduce the viewport (by resizing the browser), leaflet calculates a zoom level of 5, because zoom 6 would be too big and wouldnt fit onto the viewport.

As such, the "amount" of map that can be shown on screen depends on the viewport of the browser. So if the goal is to find a min zoom level that can fit the map, we need to take into account of the viewport size. I guess one way to do it is set a min zoom of 0 initially. When the map is loaded, leaflet will calculate a suitable zoom that will fit the map into the viewport, we then set that as the min zoom to prevent the user from zooming out too much.

@mxgrey What do you think of the proposal? I assume the initial problem is that some maps can't zoom out far enough to fit the viewport.

Reply:
That sounds good to me :+1: 

The only question it raises is whether there's ever a reason a user might want some extra zoom out, but as long as we're adjusting the minimum zoom based on the viewport size, I can't think of any reason a user would want that.

Reply:
Or maybe we could just set the min zoom to 0.

Reply:
> I guess one way to do it is set a min zoom of 0 initially.

Setting a minZoom of 0 sounds good. 

> When the map is loaded, leaflet will calculate a suitable zoom that will fit the map into the viewport, we then set that as the min zoom to prevent the user from zooming out too much.

I initially tried to use the `getZoom` method from leaflet. Unfortunately, it does not show the calculated zoom value for the new map when you are toggling between maps and gives the zoom value of the current map. This is why I resorted to the rather tedious calculations to approximate the minZoom value from width or height.

If I understand correctly, you want to somehow get the zoom value for the new map once the map changes and set it as the minZoom value without having to resort to the approximations. I think we can access it somewhere in the event. The drawback is we need to set the event type to `any`.

> To be more robust, we would also have to handle window resize events to recalculate the new min zoom.

Good Idea!

Reply:
> Or maybe we could just set the min zoom to 0.

:tada:

Reply:
Yeah, it can be a ~bug~ feature. leaflet should default to a sensible zoom level, if the user choose to zoom more too much then it is the user's choice.